### PR TITLE
Improved ComponentSerializer

### PR DESCRIPTION
**Pull request by ItsLaivy**
- Enhanced ComponentSerializer#toString methods
- Code enhanced

Now the ComponentSerializer#toString follows the default Minecraft's component json format. Multiples BaseComponents will be now serialized as a json array.

### CODE REVIEW

**Comment by md-5** [comment]
Formatting...

**Comment by ItsLaivy** [comment]
What does this exactly means?

**Comment by md-5** [comment]
Look at the diff and build errors please

**Comment by ItsLaivy** [comment]
Hahah i'm sorry, first time using this. This new commit should fix the tests now.

**Comment by ItsLaivy** [comment]
I need to do something more? It says the merging is blocked

**Comment by Janmm14** [comment]
Its only about commit signing - md-5 will resolve that by signing the commit himself, nothing you need to do

File: chat/src/main/java/net/md_5/bungee/chat/ComponentSerializer.java [hunk 1]
```diff
@@ -71,12 +80,20 @@ public static String toString(BaseComponent component)
...
         {
-            return gson.toJson( components[0] );
+            return ComponentSerializer.toString( components[0] );
         } else
         {
-            return gson.toJson( new TextComponent( components ) );
+            JsonArray array = new JsonArray();
+            for ( BaseComponent component : components )
+            {
+                array.add( JSON_PARSER.parse( ComponentSerializer.toString( component ) ) );
```
Comment (MrIvanPlays) [inline chat/src/main/java/net/md_5/bungee/chat/ComponentSerializer.java:94]: Use `gson.toJsonTree( component )` rather than toString and parse.

**Comment by md-5** [comment]
Where do you get the default as this?

Won't this break anyone (incorrectly) relying on just accessing [0]?

Re https://github.com/SpigotMC/BungeeCord/pull/3490
I think a lot of MC actually can't handle arrays now

**Comment by ItsLaivy** [comment]
What you mean exactly?

**Comment by md-5** [comment]
Why do we want arrays?

**Comment by 2008Choco** [comment]
I don't think we do want arrays, but I think this PR is necessary for the sake of consistency between `toString()` and `#parse()`.

**Comment by ItsLaivy** [comment]
> Why do we want arrays?

If you mean about this:
> Multiples BaseComponents will be now serialized as a json array.

I think it's better than the extras. It's a bit buggy, with the arrays we can easily edit the base components using the Gson parser, we will not need to access the extras, and the extras of the extras. I heavily recommend using this new way.

File: chat/src/main/java/net/md_5/bungee/chat/ComponentSerializer.java [hunk 2]
```diff
@@ -52,16 +53,24 @@ public static BaseComponent[] parse(String json)
...
             };
         }
     }
 
     public static String toString(Object object)
     {
-        return gson.toJson( object );
+        if ( object instanceof BaseComponent )
+        {
+            return toString( (BaseComponent) object );
```
Comment (Janmm14) [inline chat/src/main/java/net/md_5/bungee/chat/ComponentSerializer.java:66]: redundant

**Pull request closed without merging**